src/execution/StepExecutionManager.ts

```diff
@@ -153,6 +153,14 @@ export class StepExecutionManager {
             transactionRequest.maxPriorityFeePerGas =
               customConfig.maxPriorityFeePerGas
             transactionRequest.maxFeePerGas = customConfig.maxFeePerGas
+          } else {
+            const estimatedGasLimit = await signer.estimateGas(
+              transactionRequest
+            )
+
+            transactionRequest.gasLimit = `${Math.round(
+              Number(estimatedGasLimit) * 1.25
+            )}`
```

Comment:
Reviewing this once again, I realized that the return type from `estimateGas` is `BigNumber`, but we convert it to a plain `Number` here, and sure, it's an edge case for gas estimation, but this potentially can lead to an error if we get a big number that exceeds `Number.MAX_SAFE_INTEGER`. I think we should switch to `BigInt` here.

Also, when we switch to `viem`, this function starts to return `BigInt`, which just reinforces my words 🙂

Reply:
`${BigInt(estimatedGasLimit) * 125n / 100n}`

Reply:
Hey @chybisov,
Pushed the changes for this :)

_`/prerestart.ts`_
![CleanShot 2023-05-17 at 16 47 20](https://github.com/lifinance/sdk/assets/35174288/c3f0c609-74d1-4aca-8f0c-5e6271f91306)

_`StepExecutionManager.ts`_
![CleanShot 2023-05-17 at 16 46 50](https://github.com/lifinance/sdk/assets/35174288/e70c6486-c54f-4d07-bbc9-4930e3ca7503)